resources/js/Components/Form/EntryCreateForm.tsx

```diff
@@ -150,14 +157,27 @@ const EntryCreateForm = ({
                     id='length'
                     label='Length'
                     name='Length'
-                    type='number'
+                    InputLabelProps={{
+                        shrink: true,
+                    }}
                     InputProps={{
                         inputProps: {
                             min: 0,
+                            inputMode: 'numeric',
+                            pattern: '[0-9]*',
                         },
                     }}
                     value={data.length.toString()}
-                    onChange={(e) => setData('length', parseInt(e.target.value))}
+                    onChange={(e) => {
```

Comment:
if multiple fields have the same or almost identical `onChange` function, it is a better idea to extract them and make a re-usable component
in this case, we should probably make a re-usable component for float `textField`